quickfixj-core/src/main/java/quickfix/field/converter/UtcTimestampConverter.java

```diff
@@ -135,26 +135,37 @@ public static Date convert(String value) throws FieldConvertError {
      */
     public static LocalDateTime convertToLocalDateTime(String value) throws FieldConvertError {
         verifyFormat(value);
-        int length = value.length();
         try {
-            switch (length) {
-                case LENGTH_INCL_SECONDS:
-                    return LocalDateTime.parse(value, FORMATTER_SECONDS);
-                case LENGTH_INCL_MILLIS:
-                    return LocalDateTime.parse(value, FORMATTER_MILLIS);
-                case LENGTH_INCL_MICROS:
-                    return LocalDateTime.parse(value, FORMATTER_MICROS);
-                case LENGTH_INCL_NANOS:
-                case LENGTH_INCL_PICOS:
-                    return LocalDateTime.parse(value.substring(0, LENGTH_INCL_NANOS), FORMATTER_NANOS);
-                default:
-                    throwFieldConvertError(value, TYPE);
+            int length = value.length();
+            int ns = 0;
+            if (length >= LENGTH_INCL_NANOS) {
+                ns = parseInt(value, 18, 9);
+            } else if (length == LENGTH_INCL_MICROS) {
+                ns = parseInt(value, 18, 6) * 1000;
+            } else if (length == LENGTH_INCL_MILLIS) {
+                ns = parseInt(value, 18, 3) * 1000000;
             }
-        } catch (DateTimeParseException e) {
+    
+            int yy = parseInt(value, 0, 4);
+            int mm = parseInt(value, 4, 2);
+            int dd = parseInt(value, 6, 2);
+            int h = parseInt(value, 9, 2);
+            int m = parseInt(value, 12, 2);
+            int s = parseInt(value, 15, 2);
+            return LocalDateTime.of(yy, mm, dd, h, m, s, ns);
```

Comment:
Probably a micro optimisation, but could we have only the `LocalDateTime.of` in the try-block? It shouldn't be necessary for the `parseInt()` method.

Reply:
that make sense